### DISCUSSION TITLE
select ( 2::bit & 2::bit ) = 2::bit; FALSE?

### DISCUSSION THREAD

**Discussion by orzom411** [duckdb/duckdb#10720] [Q&A]
Good Morning All,

Bitwise operation isn't what I'd expect:
`select ( 2::bit & 2::bit ) = 2::bit as b;`

Result:
false 

Mitigation:
`select cast( ( 2::bit & 2::bit ) as int ) = cast( 2::bit as int ) as b;`

Result
true

`select 2::bit = 2::bit as b;`

Result
true

`select 2::bit & 2::bit as b;`

Result
00000000000000000000000000000010

I checked to see if this was discussed or an issue but didn't find it, sorry if I missed it.  The mitigation works fine it's just odd once a bitwise operator is involved it fails to compare.   And, oh yah, there all bits... 

`select typeof(2::bit & 2::bit)  as b;`

Result
BIT

**Comment by soerenwolfers** [comment]
https://github.com/duckdb/duckdb/issues/11211

**Comment by Mytherin** [comment]
Thanks for reporting - a fix is up here - https://github.com/duckdb/duckdb/issues/11211
